script/tool/test/pathify_command_test.dart

```diff
@@ -0,0 +1,283 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+
+import 'package:args/command_runner.dart';
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_plugin_tools/src/common/repository_package.dart';
+import 'package:flutter_plugin_tools/src/pathify_command.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import 'common/plugin_command_test.mocks.dart';
+import 'mocks.dart';
+import 'util.dart';
+
+void main() {
+  FileSystem fileSystem;
+  late Directory packagesDir;
+  late CommandRunner<void> runner;
+  late RecordingProcessRunner processRunner;
+
+  setUp(() {
+    fileSystem = MemoryFileSystem();
+    packagesDir = createPackagesDirectory(fileSystem: fileSystem);
+
+    final MockGitDir gitDir = MockGitDir();
+    when(gitDir.path).thenReturn(packagesDir.parent.path);
+    when(gitDir.runCommand(any, throwOnError: anyNamed('throwOnError')))
+        .thenAnswer((Invocation invocation) {
+      final List<String> arguments =
+          invocation.positionalArguments[0]! as List<String>;
+      // Route git calls through the process runner, to make mock output
+      // consistent with other processes. Attach the first argument to the
+      // command to make targeting the mock results easier.
+      final String gitCommand = arguments.removeAt(0);
+      return processRunner.run('git-$gitCommand', arguments);
+    });
+
+    processRunner = RecordingProcessRunner();
+    final PathifyCommand command = PathifyCommand(packagesDir, gitDir: gitDir);
+
+    runner = CommandRunner<void>('pathify_command', 'Test for $PathifyCommand');
+    runner.addCommand(command);
+  });
+
+  /// Adds dummy 'dependencies:' entries for each package in [dependencies]
+  /// to [package].
+  void _addDependencies(
+      RepositoryPackage package, Iterable<String> dependencies) {
+    final List<String> lines = package.pubspecFile.readAsLinesSync();
+    final int dependenciesStartIndex = lines.indexOf('dependencies:');
+    assert(dependenciesStartIndex != -1);
+    lines.insertAll(dependenciesStartIndex + 1, <String>[
+      for (final String dependency in dependencies) '  $dependency: ^1.0.0',
+    ]);
+    package.pubspecFile.writeAsStringSync(lines.join('\n'));
+  }
+
+  test('no-ops for no plugins', () async {
+    createFakePackage('foo', packagesDir);
+
+    final List<String> output =
+        await runCapturingPrint(runner, <String>['pathify']);
+
+    expect(
+      output,
+      containsAllInOrder(<Matcher>[
+        contains('No target packages'),
+      ]),
+    );
```

Comment:
Instead of looking at the tools output we should be looking at the pubspec file changes, no?

Reply:
The output matters too, but added a second package with a dependency on the first and validate that its pubspec is unchanged as well.